libobs/obs-output.c

```diff
@@ -875,11 +916,28 @@ void obs_output_set_audio_encoder(obs_output_t *output, obs_encoder_t *encoder,
 	output->audio_encoders[idx] = encoder;
 }
 
+obs_encoder_t *obs_output_get_video_encoder2(const obs_output_t *output,
+					     size_t idx)
+{
+	if (!obs_output_valid(output, "obs_output_get_video_encoder"))
+		return NULL;
+
+	if ((output->info.flags & OBS_OUTPUT_MULTI_TRACK_VIDEO) != 0) {
```

Comment:
Technically this logic can be simplified only to a single `if (idx >= MAX_VIDEO_ENCODERS)` because the array always exists, and the items in `idx > 0` are NULL anyway if the output is single track video, right?